Resync kni

Tal-or opened the pull request.

swatisehgal committed:
Minimize Node resource Topology scheduler plugin image

Signed-off-by: Swati Sehgal <swsehgal@redhat.com>

ffromani committed:
Merge pull request #1 from swatisehgal/minimize-scheduler-plugin-image

Minimize Node resource Topology scheduler plugin image

swatisehgal committed:
Add build-noderesourcetopology-plugin.amd64 target explicitly

Signed-off-by: Swati Sehgal <swsehgal@redhat.com>

swatisehgal committed:
Consistent binary name for noderesourcetopology plugin

In the favour of consistency and making sure that the noderesourcetopology
plugin works with the existing deployment files we keep the binary name
same as the one used for other scheduler plugins i.e kube-scheduler.

In a lot of instances, the sys admin might want to provide additional
arguments or override the existing defaults in the deployment spec and
it is better to have the binary name same as the one used in case of
other scheduler plugins.

Signed-off-by: Swati Sehgal <swsehgal@redhat.com>

ffromani committed:
Merge pull request #2 from swatisehgal/add-noderesourcetopology-amd64-target

Add build-noderesourcetopology-plugin.amd64 target explicitly

fromanirh commented:
/lgtm
/approve
this seems like the right thing to do, but I'm not experienced enough with kube scheduling to anticipate any issue. Nevertheless, let's try this out.

swatisehgal committed:
Merge pull request #3 from swatisehgal/ensure-consistent-binary-naming

Consistent binary name for noderesourcetopology plugin

ffromani committed:
Dockerfile: align with latest standards

Make it consistent with the rte-operator wherever feasible.

Signed-off-by: Francesco Romani <fromani@redhat.com>

ffromani committed:
Merge pull request #4 from openshift-kni/dockerfile-for-ocp

Dockerfile: align with latest standards

ffromani committed:
add vendor/

Signed-off-by: Francesco Romani <fromani@redhat.com>

ffromani committed:
Merge pull request #5 from openshift-kni/add-vendoring

add vendor/

ffromani committed:
noderesourcetopology-plugin: Dockerfile: remove ENTRYPOINT

We previously added the ENTRYPOINT in the Dockerfile in commit 4f30fb2 ,
but turns out this is incompatible with the configuration
the secondary-scheduler-operator injects, so we revert the change

Signed-off-by: Francesco Romani <fromani@redhat.com>

ffromani committed:
noderesourcetopology-plugin: Dockerfile: remove unused var

We don't use ARCH, so let's remove it

Signed-off-by: Francesco Romani <fromani@redhat.com>

ffromani committed:
Merge pull request #7 from openshift-kni/fix-dockerfile

Fix dockerfile

yanirq committed:
prow: add configuration

Add new OWNERS, OWNERS_ALIASES and ocp-specific dockerfiles.
and set reviewers and approvers

Signed-off-by: Yanir Quinn <yquinn@redhat.com>

ffromani committed:
Merge pull request #9 from yanirq/master

prow: add configuration

ffromani committed:
makefile: add a target to run a subset of tests

We focus on a subset of plugins, so it makes sense
to also define a subset of tests we consider tier 1
and run them first in our CI.

Signed-off-by: Francesco Romani <fromani@redhat.com>

ffromani committed:
Merge pull request #10 from openshift-kni/unit-test-quick

makefile: add a target to run a subset of tests

ffromani committed:
[KNI] Makefile: restore pristine copy, add KNI's

restore the scheduler-plugins original Makefile,
and split the KNI-specific targets and helpers in
a separate Makefile.

Signed-off-by: Francesco Romani <fromani@redhat.com>

ffromani committed:
[KNI] move hack utilities in their own directory

Let's move kni-specific hack utils in their own directory,
to facilitate future rebases.

Signed-off-by: Francesco Romani <fromani@redhat.com>

ffromani committed:
[KNI] project: fix go module references

- declare this project with its own name
  (not with name inherited from sigs.k8s.io)
- in the noderesourcetyopology-plugin, use local plugin tree

Signed-off-by: Francesco Romani <fromani@redhat.com>

ffromani committed:
[KNI] update vendoring

Signed-off-by: Francesco Romani <fromani@redhat.com>

openshift-merge-robot committed:
Merge pull request #11 from openshift-kni/cleanup-fork

Cleanup fork

Huang-Wei committed:
feat: activate sibling pods of a PodGroup

Huang-Wei committed:
fixup: address comments

Huang-Wei committed:
avoid instantiate core resource's informers

Core resource's (like Pod/Node) informers should be obtained from
frameworkHandle's SharedInformerFactory.

swatisehgal committed:
NodeResourceTopology: Update README with NRT CR dependency information

Capture the dependency information and explain that for Scheduler
plugin version > v0.21.6 depends on NRT CRD v0.0.12 and the
namespace field has been deprecated.

Signed-off-by: Swati Sehgal <swsehgal@redhat.com>

Tal-or committed:
Merge remote-tracking branch 'k8ssigstream/master' into bump_nrt_to_v0.12

Tal-or committed:
[KNI] Bump scheduler-plugins to use NRT API v0.12

 Bump to a scheduler-plugins version that works with NRT API v0.0.12

Signed-off-by: Talor Itzhak <titzhak@redhat.com>

Tal-or committed:
[KNI] Update unit-tests

Signed-off-by: Talor Itzhak <titzhak@redhat.com>

openshift-merge-robot committed:
Merge pull request #12 from Tal-or/bump_nrt_to_v0.12

[KNI] Bump scheduler-plugins to use NRT API v0.12

k8s-ci-robot committed:
Merge pull request #304 from k8stopologyawareschedwg/nrt-doc-update

NodeResourceTopology: Update README with NRT CR dependency information

k8s-ci-robot committed:
Merge pull request #302 from Huang-Wei/rm-extra-informer

avoid instantiate core resource's informers

spiffxp committed:
images: use k8s-staging-test-infra/gcb-docker-gcloud

k8s-ci-robot committed:
Merge pull request #305 from spiffxp/use-k8s-infra-for-gcb-image

images: use k8s-staging-test-infra/gcb-docker-gcloud

spiffxp committed:
images: fix invalid k8s-staging-test-infra/gcb-docker-gcloud tag

k8s-ci-robot committed:
Merge pull request #306 from spiffxp/fix-gcb-image

images: fix invalid k8s-staging-test-infra/gcb-docker-gcloud tag

k8s-ci-robot committed:
Merge pull request #299 from Huang-Wei/activate-podgroup-pods

feat: activate sibling pods of a PodGroup

Tal-or committed:
[KNI] refer to specific Makefile in Dockerfile

We should specifiy the correct makefile in the dockerfile since the default is the makefile named Makefile which doesn't have the build-noderesourcetopology-plugin make target

Signed-off-by: Talor Itzhak <titzhak@redhat.com>

openshift-merge-robot committed:
Merge pull request #14 from Tal-or/fix_dockerfile

[KNI] refer to specific Makefile in Dockerfile

Tal-or committed:
Merge remote-tracking branch 'oldknistream/master' into resync_kni
